### ISSUE TITLE
AIOBE during layout with HierarchyHandling.INCLUDE_CHILDREN

### ISSUE THREAD

**Issue by JanKoehnlein** [eclipse/elk#226]
[model.txt](https://github.com/eclipse/elk/files/1245499/model.txt)

```
 java.lang.ArrayIndexOutOfBoundsException: -1
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.CrossingMatrixFiller.<init>(CrossingMatrixFiller.java:38)
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.GreedySwitchHeuristic.getNewSwitchDecider(GreedySwitchHeuristic.java:87)
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.GreedySwitchHeuristic.setUp(GreedySwitchHeuristic.java:82)
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.GreedySwitchHeuristic.setFirstLayerOrder(GreedySwitchHeuristic.java:74)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.minimizeCrossingsNoCounter(LayerSweepCrossingMinimizer.java:152)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.lambda$1(LayerSweepCrossingMinimizer.java:120)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.minimizeCrossings(LayerSweepCrossingMinimizer.java:129)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.process(LayerSweepCrossingMinimizer.java:108)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.process(LayerSweepCrossingMinimizer.java:1)
	at org.eclipse.elk.alg.layered.ElkLayered.hierarchicalLayout(ElkLayered.java:234)
	at org.eclipse.elk.alg.layered.ElkLayered.doCompoundLayout(ElkLayered.java:180)
	at org.eclipse.elk.alg.layered.LayeredLayoutProvider.layout(LayeredLayoutProvider.java:58)
	at org.eclipse.elk.core.RecursiveGraphLayoutEngine.layoutRecursively(RecursiveGraphLayoutEngine.java:181)
	at org.eclipse.elk.core.RecursiveGraphLayoutEngine.layout(RecursiveGraphLayoutEngine.java:60)
```

**Comment by uruuru** [comment]
It's a known problem: #155. 

Workaround: `org.eclipse.elk.layered.crossingMinimization.greedySwitch.type: OFF`

**Comment by JanKoehnlein** [comment]
Your workaround yields a (broken) purely horizontal or purely vertical layout without of all diagrams:

[broken_layout.txt](https://github.com/eclipse/elk/files/1248722/broken_layout.txt)

**Comment by uruuru** [comment]
Should be another unrelated issue. I created a new bug for it. Thanks.

**Comment by uruuru** [comment]
It should help if you switch greedy switch off for _every_ hierarchical node.

**Comment by JanKoehnlein** [comment]
What exactly is a hierarchical node?

**Comment by uruuru** [comment]
Every node that has further child-nodes: [terminology](http://www.eclipse.org/elk/documentation/tooldevelopers/graphdatastructure.html).

**Comment by JanKoehnlein** [comment]
That yields an AIOOBE again

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: -1
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.CrossingMatrixFiller.<init>(CrossingMatrixFiller.java:38)
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.GreedySwitchHeuristic.getNewSwitchDecider(GreedySwitchHeuristic.java:87)
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.GreedySwitchHeuristic.setUp(GreedySwitchHeuristic.java:82)
	at org.eclipse.elk.alg.layered.intermediate.greedyswitch.GreedySwitchHeuristic.setFirstLayerOrder(GreedySwitchHeuristic.java:74)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.minimizeCrossingsNoCounter(LayerSweepCrossingMinimizer.java:152)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.lambda$1(LayerSweepCrossingMinimizer.java:120)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.minimizeCrossings(LayerSweepCrossingMinimizer.java:129)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.process(LayerSweepCrossingMinimizer.java:108)
	at org.eclipse.elk.alg.layered.p3order.LayerSweepCrossingMinimizer.process(LayerSweepCrossingMinimizer.java:1)
	at org.eclipse.elk.alg.layered.ElkLayered.hierarchicalLayout(ElkLayered.java:234)
	at org.eclipse.elk.alg.layered.ElkLayered.doCompoundLayout(ElkLayered.java:180)
	at org.eclipse.elk.alg.layered.LayeredLayoutProvider.layout(LayeredLayoutProvider.java:58)
	at org.eclipse.elk.core.RecursiveGraphLayoutEngine.layoutRecursively(RecursiveGraphLayoutEngine.java:181)
	at org.eclipse.elk.core.RecursiveGraphLayoutEngine.layout(RecursiveGraphLayoutEngine.java:60)
```
[model.txt](https://github.com/eclipse/elk/files/1250041/model.txt)

**Comment by JanKoehnlein** [comment]
Aha, if I set it for the root and the hierarchical nodes, it seems to work.

**Comment by uruuru** [comment]
The root is a hierarchical node itself :). I believe I understood the issue, should be fixed soon #228.
